src/components/ShareResultsModal/EmailField.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import * as z from 'zod';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import CheckMarkCircleSVG from './CheckMarkCircleSVG';
+
+const emailSchema = z.object({
+  email: z.string().email({ message: 'Please enter a valid email address' }),
+});
+
+export default function EmailField({ onSubmit }: Props) {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitSuccessful },
+  } = useForm<{ email: string }>({
+    resolver: zodResolver(emailSchema),
+  });
+
+  return (
+    <div className='cio-share-results-feature-group'>
+      <form onSubmit={handleSubmit(({ email }) => onSubmit(email))}>
+        <div className='cio-share-results-description'>Share by email</div>
+        <div className='cio-share-results-button-group'>
+          <div className='cio-share-results-email-input-group'>
+            <input
+              className={`cio-share-results-email-input ${
+                errors.email ? 'cio-share-results-email-input--error' : ''
+              }`}
+              {...register('email')}
+            />
+            {errors.email && (
+              <div className='cio-share-results-email-input-error-message'>
+                {errors.email.message}
+              </div>
+            )}
+          </div>
+          <button className='cio-share-results-share-button' type='submit'>
+            Send
+          </button>
+        </div>
+      </form>
+      {isSubmitSuccessful && (
+        <div className='cio-share-results-copied-notification'>
+          <CheckMarkCircleSVG />
+          <div>Email sent</div>
+        </div>
+      )}
+    </div>
+  );
+}
+
+type Props = {
```

Comment:
Not a big deal but it would be better if we follow the existing patterns. We usually put the types to the top of the file as an interface. Can we do that just to keep it consistent? Tagging TS wizard @mocca102 

https://github.com/Constructor-io/constructorio-ui-quizzes/blob/bedaf3e82e855237e2dcef1854a1846c6864a82b/src/components/RedoButton/RedoButton.tsx#L5

Reply:
Whoops, good catch! Sorry, I should've aligned with the existing patterns instead of borrowing practices from another 😛 27f85b661c45a8c98ce347cd87ae97e484c8900e

Reply:
Whoops, good catch! Sorry, I should've aligned with the existing patterns instead of borrowing practices from another 😛 27f85b661c45a8c98ce347cd87ae97e484c8900e

Reply:
Whoops, good catch! Sorry, I should've aligned with the existing patterns instead of borrowing practices from another 😛 75b49eb303d05807142008881e97dc49fef9d339